frontend/pages/beta/model/[modelId]/schema.tsx

```diff
@@ -1,23 +1,34 @@
 import { Schema } from '@mui/icons-material'
 import { Button, Card, Grid, Stack, Tooltip, Typography } from '@mui/material'
+import _ from 'lodash-es'
 import { useRouter } from 'next/router'
 import { useMemo } from 'react'
 
+import { useGetModel } from '../../../../actions/model'
+import { postFromSchema } from '../../../../actions/modelCard'
 import { useGetSchemas } from '../../../../actions/schema'
+import { useGetCurrentUser } from '../../../../actions/user'
 import EmptyBlob from '../../../../src/common/EmptyBlob'
 import Loading from '../../../../src/common/Loading'
+import MessageAlert from '../../../../src/MessageAlert'
 import Wrapper from '../../../../src/Wrapper.beta'
+import { SchemaInterface } from '../../../../types/types'
 
 export default function NewSchemaSelection() {
   const router = useRouter()
   const { modelId }: { modelId?: string } = router.query
   const { schemas, isSchemasLoading, isSchemasError } = useGetSchemas()
+  const { model, isModelLoading, isModelError } = useGetModel(modelId)
+  const { currentUser, isCurrentUserLoading, isCurrentUserError } = useGetCurrentUser()
 
   const activeSchemas = useMemo(() => schemas.filter((schema) => schema.active), [schemas])
   const inactiveSchemas = useMemo(() => schemas.filter((schema) => !schema.active), [schemas])
 
-  function createModelUsingSchema(_schema: string) {
-    router.push(`/beta/model/${modelId}`)
+  async function createModelUsingSchema(newSchema: SchemaInterface) {
+    if (currentUser && model) {
+      await postFromSchema(model.id, newSchema.id)
+      router.push(`/beta/model/${modelId}`)
+    }
   }
 
   function schemaButton(schema: any) {
```

Comment:
Looks like this slipped through a previous PR, but this is essentially a component pretending to be a regular function. Up to you if you want to fix it in this PR though